hack/check-skaffold-deps.sh

```diff
@@ -18,76 +18,21 @@ set -e
 
 echo "> Check Skaffold Dependencies"
 
-check_successful=true
-
-out_dir=$(mktemp -d)
-function cleanup_output {
-  rm -rf "$out_dir"
-}
-trap cleanup_output EXIT
-
-function check() {
-  skaffold_file="$1"
-  binary_name="$2"
-  skaffold_config_name="$3"
-
-  skaffold_yaml="$(cat "$(dirname "$0")/../$skaffold_file")"
-
-  path_current_skaffold_dependencies="${out_dir}/current-$skaffold_file-deps-$binary_name.txt"
-  path_actual_dependencies="${out_dir}/actual-$skaffold_file-deps-$binary_name.txt"
-
-  echo "$skaffold_yaml" |\
-    yq eval "select(.metadata.name == \"$skaffold_config_name\") | .build.artifacts[] | select(.ko.main == \"./cmd/$binary_name\") | .ko.dependencies.paths[]?" - |\
-    sort |\
-    uniq > "$path_current_skaffold_dependencies"
-
-  go list -f '{{ join .Deps "\n" }}' "./cmd/$binary_name" |\
-    grep "github.com/gardener/gardener/" |\
-    sed 's/github\.com\/gardener\/gardener\///g' |\
-    sort |\
-    uniq > "$path_actual_dependencies"
-
-  # always add vendor directory and VERSION file
-  echo "vendor" >> "$path_actual_dependencies"
-  echo "VERSION" >> "$path_actual_dependencies"
-
-  # sort dependencies
-  sort -o $path_current_skaffold_dependencies{,}
-  sort -o $path_actual_dependencies{,}
-
-  echo -n ">> Checking defined dependencies in Skaffold config '$skaffold_config_name' for '$binary_name' in '$skaffold_file'..."
-  if ! diff="$(diff "$path_current_skaffold_dependencies" "$path_actual_dependencies")"; then
-    check_successful=false
-
-    echo
-    echo ">>> The following actual dependencies are missing in $skaffold_file (need to be added):"
-    echo "$diff" | grep '>' | awk '{print $2}'
-    echo
-    echo ">>> The following dependencies defined in $skaffold_file are not needed actually (need to be removed):"
-    echo "$diff" | grep '<' | awk '{print $2}'
-    echo
-  else
-    echo " success."
-  fi
-}
+REPO_ROOT="$(git rev-parse --show-toplevel)"
 
 # skaffold.yaml
-check "skaffold.yaml" "gardener-admission-controller"      "controlplane"
-check "skaffold.yaml" "gardener-apiserver"                 "controlplane"
-check "skaffold.yaml" "gardener-controller-manager"        "controlplane"
-check "skaffold.yaml" "gardener-extension-provider-local"  "provider-local"
-check "skaffold.yaml" "gardener-resource-manager"          "gardenlet"
-check "skaffold.yaml" "gardener-scheduler"                 "controlplane"
-check "skaffold.yaml" "gardenlet"                          "gardenlet"
+$REPO_ROOT/hack/check-skaffold-deps-for-binary.sh --skaffold-file "skaffold.yaml" --binary "gardener-admission-controller"     --skaffold-config "controlplane"
```

Comment:
If we keep the `check` method and move the invocation of `$REPO_ROOT/hack/check-skaffold-deps-for-binary.sh` there we could preserve the old behaviour where all binaries are checked. Currently the first occurrence of a dependency mismatch will cause the  script to exit so it might be necessary to be executed multiple times.

What I have in mind is something like this:


```bash
check_successful=true

function check() {
  if ! $REPO_ROOT/hack/check-skaffold-deps-for-binary.sh --skaffold-file $1 --binary $2 --skaffold-config $3; then
    check_successful=false
  fi
}

check "skaffold.yaml" "gardener-admission-controller"      "controlplane"
...
...

if [ "$check_successful" = false ] ; then
  exit 1
fi
```

Wdyt?

Reply:
Thanks! That's a nice trick. I was wondering how to keep the old behaviour but was not able to find out an good approach.